File: Ocean Scream最终版/Ocean Scream V1.0/Ocean Scream游戏平台/AIdemo/C#/AI.cs
using System;
using System.Collections.Generic;
using System.Text;
using MSTCOS.SDK;

namespace MSTCOS.SDK
{
    public class AI : AbstractAI
    {
        public AI(OSInterface Interface)
            : base(Interface)
        {
        }

        List<ShipInfo> ships, a, b;
        Random random = new Random();

        private float dist(ShipInfo a, ShipInfo b)
        {
            return (a.PositionX - b.PositionX) * (a.PositionX - b.PositionX) + (a.PositionY - b.PositionY) * (a.PositionY - b.PositionY);
        }

        public override void Iteration()
        {
            int i, j, k;
            float d, t;

            Interface.Data();
            ships = Interface.Ship;

            a = new List<ShipInfo>();
            b = new List<ShipInfo>();

            for (i = 0; i < ships.Count; i++)
                if (ships[i].Faction == Interface.Faction) a.Add(ships[i]);
                else b.Add(ships[i]);

            for (i = 0; i < a.Count; i++)
            {
                d = -1;
                k = 0;
                for (j = 0; j < b.Count; j++)
                {
                    t = dist(a[i], b[j]);
                    if (t < d || d == -1)
                    {
                        d = t;
                        k = j;
                    }
                }
                if (b.Count > 0)
                {
                    Interface.MoveTo(a[i].ID, b[k].PositionX + (float)random.NextDouble() * 200 - 100, b[k].PositionY + (float)random.NextDouble() * 200 - 100);
                    if (d < OSInterface.CannonRange * OSInterface.CannonRange) Interface.Attack(a[i].ID, b[k].ID);
                }
                else Interface.StartMoving(a[i].ID);
            }

        }

        public static void Main(String[] args)
        {
            String[] temp = new String[2];
            temp[0] = "buaacs";
            temp[1] = "111.222.33";
            SDK.StartGame(temp);
        }
    }
}

File: Ocean Scream最终版/Ocean Scream V1.0/Ocean Scream游戏平台/AIdemo/C++/AI/AI.cpp
#include "MSTCOSSDK.h"
#include <iostream>
#include <cmath>
#include <cstring>
#include <algorithm>
#pragma comment(lib,"MSTCOSSDK.lib")
using namespace std;

vector<ShipInfo> ships,a,b;

float dist(ShipInfo a,ShipInfo b)
{
    return (a.PositionX - b.PositionX) * (a.PositionX - b.PositionX) + (a.PositionY - b.PositionY) * (a.PositionY - b.PositionY);
}

void Iteration(OSInterface* const Interface)
{
    int i, j, k;
    float t, d;
    Interface->Data();
    ships = Interface->Ship;
    a.clear();
    b.clear();
    for (i = 0; i<ships.size(); i++)
    {
        if (ships[i].Faction == Interface->Faction) a.push_back(ships[i]);
        else b.push_back(ships[i]);
    }

    for (i = 0; i < a.size(); i++)
    {
        d = -1;
        k = 0;
        for (j = 0; j < b.size(); j++)
        {
            t = dist(a[i], b[j]);
            if (t < d || d == -1)
            {
                d = t;
                k = j;
            }
        }
        if (b.size() > 0)
        {
            Interface->MoveTo(a[i].ID, b[k].PositionX+rand()%200-100, b[k].PositionY+rand()%200-100);
            if (d < CannonRange * CannonRange) Interface->Attack(a[i].ID, b[k].ID);
        }
        else Interface->StartMoving(a[i].ID);
    }
}

int main(int argc, char* argv[])
{
    char* temp[2];
    temp[0] = "buaacpp";
    temp[1] = "11.22.33";
    srand(0);
    MSTCOSSDK::StartGame(2, temp, Iteration);
    return 0;
}